validator/validation/readers.py

```diff
@@ -188,7 +241,14 @@ def create_reader(dataset, version) -> GriddedNcTs:
 
     if dataset.short_name == globals.SMAP_L2:
         reader = GriddedNcOrthoMultiTs(folder_name, ioclass_kws={'read_bulk': True})
-
+        # and version.short_name == globals.SMAP_V9_AM_PM
+    if dataset.short_name == globals.SMAP_L3 and version.short_name == 'SMAP_V9_AM_PM':
```

Comment:
looks like we are duplicating smap_data_folder definition, but I guess that's a task for general code improvement